theano/compile/function_module.py

```diff
@@ -676,6 +676,18 @@ def __call__(self, *args, **kwargs):
         elif self.unpack_single and len(outputs) == 1:
             return outputs[0]
         else:
+
+            if self.output_keys is not None: 
+                outputDict = {}
+
+
+                assert len(self.output_keys) == len(outputs)
+
+                #for i in range(0, len(self.output_keys)): 
+                #    outputDict[self.output_keys[i]] = outputs[i]
+
+                return dict(itertools.izip(self.output_keys, outputs))
```

Comment:
ok. it is great that you timmed it. You can be surprised of where overhead come from. For small Theano function like on scalar, the parsing of the input is a significant overhead...